### PR TITLE
feat(a2ui): add headless message buffer and surface processor

### PR DESCRIPTION
**Pull request by PupilTong**
## Summary

Foundational message-buffer + payload-normalization layer for the headless A2UI renderer. Pure data logic, no React surface, no breaking changes.

- `MessageStore` — append-only buffer of raw protocol messages with a `useSyncExternalStore`-friendly `subscribe` / `getSnapshot` API.
- `payloadNormalizer` — turns arbitrary developer payloads (string, structured object, `{ kind, data }` envelope, nested arrays) into a flat `ServerToClientMessage[]`. Includes `createFallbackMessagesFromPlainText`, `createTextCardMessages`, and `prepareMessagesForProcessing` helpers used by IO transports.

Adds a per-package `rstest.config.ts` plus split `tsconfig.json` (lint + tests) / `tsconfig.build.json` (emit) so tests run via `pnpm -F @lynx-js/a2ui-reactlynx test` without polluting the emit `rootDir`.

This PR is **purely additive** — the existing `core/`, `chat/`, and `utils/` paths (`BaseClient`, `A2UIRender`, the global `componentRegistry`, the legacy `createResource` / `SignalStore`) keep working untouched. The bigger types-divergent pieces (`MessageProcessor`, the new `Resource` with `subscribe` / `getSnapshot` / explicit status, the new `SignalStore`) ship together with the React renderer in [#2571](https://github.com/lynx-family/lynx-stack/pull/2571), since they introduce a new `Surface` shape that the catalog components must migrate to in the same PR.

## Test plan

- [ ] \`pnpm -F @lynx-js/a2ui-reactlynx test\` — 16/16 pass (`MessageStore`, `payloadNormalizer`)
- [ ] \`pnpm -F @lynx-js/a2ui-reactlynx build\` — extractor still emits the 10 catalog manifests
- [ ] Existing playground (which uses `core/`/`chat/` paths) still runs

## Summary by CodeRabbit

* **New Features**
  * Added a message store and payload-normalization utilities for handling and preparing runtime messages.

* **Tests**
  * Added tests covering message store behavior, payload normalization, message processing, and edge cases.

* **Chores**
  * Integrated a test runner and dev-dependency, and updated TypeScript build/test configurations and project references.

### CODE REVIEW

**Review by Sherry-hue — DISMISSED** [review]

File: packages/genui/a2ui/src/store/MessageProcessor.ts [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+  message: Record<string, unknown>;
+  resolve: (response: unknown) => void;
+}
+
+function isObject(value: unknown): value is Record<string, unknown> {
+  return typeof value === 'object' && value !== null;
+}
+
+export class MessageProcessor {
+  surfaces: Map<string, Surface>;
```
[the remark this answers is not in the document]
Comment (PupilTong) [reply packages/genui/a2ui/src/store/MessageProcessor.ts]: Stale — `MessageProcessor.ts` was moved out of this PR (PR-α) and now ships in PR-β [#2571](https://github.com/lynx-family/lynx-stack/pull/2571). Will address there.

File: packages/genui/a2ui/src/store/MessageProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+      }
+    }
+
+    if (Array.isArray(anyCloned['children'])) {
+      anyCloned['children'] = newChildren;
+    } else if (newChildren.length > 0) {
+      anyCloned['children'] = newChildren;
+    }
+
+    return newId;
```
[the remark this answers is not in the document]
Comment (PupilTong) [reply packages/genui/a2ui/src/store/MessageProcessor.ts]: Stale — `MessageProcessor.ts` was moved out of this PR (PR-α) and now ships in PR-β [#2571](https://github.com/lynx-family/lynx-stack/pull/2571). Will address the `cloneComponentTree` ref-update issue there.

File: packages/genui/a2ui/src/store/payloadNormalizer.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        }
+        return;
+      }
+
+      if (Array.isArray(v['messages'])) {
+        handle(v['messages']);
+        return;
+      }
+
+      add(createFallbackMessagesFromPlainText(JSON.stringify(v)));
```
[the remark this answers is not in the document]
Comment (PupilTong) [reply packages/genui/a2ui/src/store/payloadNormalizer.ts]: Fixed in 5012592e — wrapped the `JSON.stringify` fallback in try/catch and substitute `'[unserializable object]'` when serialization throws (circular refs, BigInt fields).

**Review by Sherry-hue — APPROVED** [review]
